tools/dart_tools/bin/update_flutter_engine.dart

```diff
@@ -0,0 +1,241 @@
+}
+
+final platformInfo = {
+  'linux':
+      new PlatformInfo('linux-x64/linux-x64-embedder', 'libflutter_engine.so'),
+  'macos': new PlatformInfo(
+      'darwin-x64/FlutterEmbedder.framework.zip', 'FlutterEmbedder.framework'),
+  'windows': new PlatformInfo(
+      'windows-x64/windows-x64-embedder.zip', 'flutter_engine.dll'),
+};
+
+main(List<String> arguments) async {
+  var parser = new ArgParser();
+  parser.addOption('platform',
+      help: 'The platform to download the engine for.\n'
+          'Defaults to the current platform.',
+      allowed: platformInfo.keys,
+      defaultsTo: Platform.operatingSystem);
+  parser.addOption('flutter_root',
+      help: 'The root of a the Flutter tree to get the engine version from.\n'
+          'Ignored if --hash is provided.\n'
+          'Defaults to a "flutter" directory next to this repository.',
+      defaultsTo: path.join(path.dirname(getRepositoryParent()), 'flutter'));
+  parser.addOption(
+    'hash',
+    help: 'The hash of the engine version to use.\n'
+        'This is only required if you want to override the version;\n'
+        'normally you should use flutter_root instead.',
+  );
+  parser.addFlag('help', help: 'Prints this usage message.', negatable: false);
+  ArgResults parsedArguments;
+  try {
+    parsedArguments = parser.parse(arguments);
+  } catch (_) {
+    printUsage(parser);
+    exit(1);
+  }
+
+  if (parsedArguments['help'] || parsedArguments.rest.length != 1) {
+    printUsage(parser);
+    exit(parsedArguments['help'] ? 0 : 1);
+  }
+
+  try {
+    final platform = parsedArguments['platform'];
+    final outputRoot =
+        path.canonicalize(path.absolute(parsedArguments.rest[0]));
+    final targetHash = parsedArguments['hash'] ??
+        await engineHashForFlutterTree(parsedArguments['flutter_root']);
+    final libraryFile = platformInfo[platform].libraryFile;
+
+    final currentHash = await lastDownloadedEngineHash(outputRoot, platform);
+    if (currentHash == null || targetHash != currentHash) {
+      await downloadEngine(targetHash, platform, outputRoot);
+      await setLastDownloadedEngineHash(outputRoot, targetHash);
+      print('Downloaded $libraryFile version ${targetHash}.');
+    } else {
+      print('$libraryFile version ${targetHash} already present.');
+    }
+  } catch (e) {
+    print(e);
+    exit(1);
+  }
+}
+
+printUsage(ArgParser argParser) {
+  print('Usage: update_flutter_engine [options] <output directory>\n');
+  print(argParser.usage);
+}
+
+// Returns the path to the parent directory of this project's repository.
+// Relies on the known location of this script within the repo.
+String getRepositoryParent() {
+  final scriptUri = Platform.script;
+  return new File.fromUri(scriptUri).parent.parent.parent.parent.path;
+}
+
+// Returns the engine version hash for the Flutter tree at [flutterRoot],
+// or null if it can't be found.
+engineHashForFlutterTree(String flutterRoot) async {
```

Comment:
These async functions should have a return type.  If they're `void` the convention is to set the return value as `Future<Null>`+

Reply:
Done (using the apparently-now-supported `Future<void>`)